cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -0,0 +1,142 @@
+    prov_data = provider.data['provisioning']
+    prov_form_data = {
+        'request': {'email': fauxfactory.gen_email(),
+                    'first_name': fauxfactory.gen_alpha(),
+                    'last_name': fauxfactory.gen_alpha()},
+        'catalog': {'num_vms': '1',
+                    'vm_name': fauxfactory.gen_alpha()},
+        'environment': {'cloud_network': prov_data['cloud_network'],
+                        'cloud_tenant': prov_data['cloud_tenant']},
+        'properties': {'instance_type': partial_match(flavor_name)},
+    }
+
+    instance_name = prov_form_data['catalog']['vm_name']
+
+    try:
+        instance = provider.appliance.collections.cloud_instances.create(
+            instance_name,
+            provider,
+            prov_form_data, find_in_cfme=True
+        )
+
+    except KeyError:
+        # some yaml value wasn't found
+        pytest.skip('Unable to find an image map in provider "{}" provisioning data: {}'
+                    .format(provider, prov_data))
+
+    yield instance
+    try:
+        instance.cleanup_on_provider()
+    except Exception:
+        pass
+
+
+def test_create_instance_with_zero_disk_flavor(new_instance, soft_assert):
+    view = navigate_to(new_instance, 'Details')
+    prov_data = new_instance.provider.data['provisioning']
+    power_state = view.entities.summary('Power Management').get_text_of('Power State')
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = view.entities.summary('Relationships').get_text_of('VM Template')
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    flavors = [f.name for f in new_instance.provider.mgmt.api.flavors.list()]
+    flavor = view.entities.summary('Relationships').get_text_of('Flavor')
+    soft_assert(flavor in flavors)
+
+    # Assert other relationships in a loop
+    props = [('Availability Zone', 'availability_zone'),
+             ('Cloud Tenants', 'cloud_tenant')]
+
+    if current_version() >= '5.7':
+        props.append(('Virtual Private Cloud', 'cloud_network'))
+
+    for p in props:
+        v = view.entities.summary('Relationships').get_text_of(p[0])
+        soft_assert(v == prov_data[p[1]])
+
+
+def test_flavor_crud(appliance, provider):
+    collection = appliance.collections.cloud_flavors
+    flavor = collection.create(name=fauxfactory.gen_alpha(),
+                               provider=provider,
+                               ram=RAM,
+                               vcpus=VCPUS,
+                               disk=DISK_SIZE,
+                               swap=SWAP_SIZE,
+                               rxtx=RXTX)
+
+    view = appliance.browser.create_view(navigator.get_class(collection, 'All').VIEW)
+    view.flash.assert_success_message(
+        'Add of Flavor "{}" was successfully initialized.'.format(flavor.name))
+
+    wait_for(lambda: flavor.exists, delay=5, timeout=600, fail_func=flavor.refresh,
+             message='Wait for flavor to appear')
+
+    assert flavor.exists
```

Comment:
Note wait_for returns a tuple of the result of the function from the last call, and the time waited. You can capture that tuple and use it instead of calling exists again:

```
flavor_exists, _ = wait_for(<etc>)
assert flavor_exists
```

But its kind of a moot assert, because if it is False, wait_for would have timed out.